lib/tcx.js

```diff
@@ -0,0 +1,207 @@
+    cadence: cadence ? parseFloat(nodeVal(cadence)) : null,
+  };
+  const extensions = x.getElementsByTagNameNS(
+    "http://www.garmin.com/xmlschemas/ActivityExtension/v2",
+    "*"
+  );
+  if (extensions !== null) {
+    for (let i = 0; i < extensions.length; i++) {
+      const extension = extensions[i];
+      attributeNames
+        .map((r) => r[0])
+        .filter((n) => n === extension.localName)
+        .forEach((name) => {
+          const raw = nodeVal(extension);
+          if (raw !== null) {
+            const v = parseFloat(raw);
+            if (!isNaN(v)) {
+              result[name] = v;
+            }
+          }
+        });
+    }
+  }
+  return result;
+}
+
+function getPoints(node, pointname) {
+  const pts = node.getElementsByTagName(pointname);
+  const line = [];
+  const times = [];
+  const l = pts.length;
+  const extendedValues = {};
+  if (l < 2) return {}; // Invalid line in GeoJSON
+  for (let i = 0; i < l; i++) {
+    const c = coordPair(pts[i]);
+    if (c === null) continue;
+    line.push(c.coordinates);
+    if (c.time) times.push(c.time);
+    attributeNames
+      .map((r) => r[0])
+      .forEach((name) => {
+        if (c[name] || extendedValues[name]) {
+          if (!extendedValues[name]) {
+            extendedValues[name] = Array(i).fill(null);
+          }
+          extendedValues[name].push(c[name] || null);
+        }
+      });
+  }
+  const result = {
+    line: line,
+    times: times,
+  };
+  attributeNames.forEach((n) => {
+    if (extendedValues[n[0]]) {
+      result[n[1]] = extendedValues[n[0]] || [];
+    }
+  });
+  return result;
+}
+
+function getLap(node) {
+  const segments = node.getElementsByTagName("Track");
+  const track = [];
+  const times = [];
+  const extendedValues = {};
+  let line;
+  for (let i = 0; i < segments.length; i++) {
+    line = getPoints(segments[i], "Trackpoint");
+    if (line) {
+      if (line.line) track.push(line.line);
+      if (line.times && line.times.length) times.push(line.times);
+
+      attributeNames
+        .map((r) => r[1])
+        .forEach((name) => {
+          if (
+            (extendedValues[name] && extendedValues[name].length) ||
+            (line[name] && line[name].length)
+          ) {
+            if (!extendedValues[name]) {
+              extendedValues[name] = [];
+            }
+            if (!extendedValues[name].length) {
+              for (let s = 0; s < i; s++) {
+                extendedValues[name].push(Array(track[s].length).fill(null));
+              }
+            }
+            if (line[name] && line[name].length) {
+              extendedValues[name].push(line[name]);
+            } else {
+              extendedValues[name].push(
+                Array(line.line.length || 0).fill(null)
+              );
+            }
+          }
+        });
+    }
+  }
+  if (track.length === 0) return;
+  const properties = Object.assign(
+    getProperties(node)
+  );
```

Comment:
This line doesn't do anything, as far as I can tell - we're not merging properties or making a clone of the object. Most likely the `Object.assign` can be removed and we're assigning `properties` to the return value of `getProperties`